misc/dist/openharmony_template/entry/src/main/ets/pages/Index.ets

```diff
@@ -0,0 +1,526 @@
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+import plugin, { SimplifiedKeyEvent, SimplifiedMouseEvent } from 'libentry.so';
+import { resourceManager } from '@kit.LocalizationKit';
+import { display } from '@kit.ArkUI';
+import { abilityAccessCtrl, common, Permissions } from '@kit.AbilityKit';
+import { KeyCode } from '@kit.InputKit';
+
+class PluginXComponentController extends XComponentController {
+  ctx: common.UIAbilityContext
+  resMgr: resourceManager.ResourceManager
+  windowId: number
+
+  constructor(context: common.UIAbilityContext, resourceManager: resourceManager.ResourceManager) {
+    super()
+    this.ctx = context
+    this.resMgr = resourceManager
+    this.windowId = this.ctx.windowStage.getMainWindowSync()
+      .getWindowProperties()
+      .id
+  }
+
+  onSurfaceCreated(surfaceId: string): void {
+    plugin.setResourceManager(this.resMgr)
+    plugin.setWindowId(this.windowId)
+    plugin.setSurfaceId(BigInt(surfaceId))
+  }
+
+  onSurfaceChanged(surfaceId: string, rect: SurfaceRect): void {
+    plugin.changeSurface(BigInt(surfaceId), rect.surfaceWidth, rect.surfaceHeight)
+  }
+
+  onSurfaceDestroyed(surfaceId: string): void {
+    plugin.destroySurface(BigInt(surfaceId))
+  }
+}
+
+@Entry
+@Component
+struct Index {
+  resMgr = this.getUIContext().getHostContext()!.resourceManager
+  context = getContext(this) as common.UIAbilityContext
```

Comment:
When using `hvigorw` with a higher API level, a warning "'getContext' has been deprecated" will be output.

https://developer.huawei.com/consumer/en/doc/harmonyos-references/js-apis-getcontext#getcontextdeprecated